src/lib/updateGrantsRecipients.js

```diff
@@ -399,7 +399,7 @@ export async function processFiles(hashSumHex) {
           && grantIds.includes(parseInt(g.replacement_grant_award_id, 10)),
       );
 
-      const grantReplacementPromises = grantsToUpdate.map(async (g) => {
```

Comment:
based on the data in the DB, it does not look like a race condition, it looks like it never checks if the value already exists

Reply:
after further investigation i was wrong, it was a race condition within the same transaction.